File: netcore/src/Zcy.Entity/Company/SupplierClient.cs
using Zcy.BaseInterface.Entities;

namespace Zcy.Entity.Company
{
    /// <summary>
    /// 供应商|客户信息
    /// </summary>
    public class SupplierClient : BaseEntity<long>, IBaseCompany
    {
        /// <summary>
        /// 客户信息
        /// </summary>
        /// <param name="clientName">客户名</param>
        /// <param name="clientType">客户类型</param>
        public SupplierClient(string clientName, ClientTypeEnum clientType)
        {
            ClientName = clientName;
            ClientType = clientType;
            ClientStatus = PublicStatusEnum.Normal;
        }


        /// <summary>
        /// 客户名称
        /// </summary>
        public string ClientName { get; protected set; }

        /// <summary>
        /// 客户状态
        /// </summary>
        public PublicStatusEnum ClientStatus { get; protected set; }

        /// <summary>
        /// 客户类型
        /// </summary>
        public ClientTypeEnum ClientType { get; protected set; }

        /// <summary>
        ///  联系方式
        /// </summary>
        public string? PhoneNumber { get; set; }

        /// <summary>
        /// 备注
        /// </summary>
        public string? Remark { get; set; }

        /// <summary>
        /// 公司Id
        /// </summary>
        public long CompanyId { get; set; }

        public void SetClientName(string clientName)
        {
            ClientName = clientName;
        }

        /// <summary>
        /// 禁用
        /// </summary>
        public void Ban()
        {
            ClientStatus = PublicStatusEnum.Ban;
        }

        /// <summary>
        /// 启用
        /// </summary>
        public void Open()
        {
            ClientStatus = PublicStatusEnum.Normal;
        }
    }
}

File: h5.vue/src/services/supplierClient.ts
import type { GetValidSupplierClientDto } from '@/types/company'
import { http } from '@/utils/http'
import { PublicStatusEnum, type ClientTypeEnum } from './constants'
import type { PageResult } from '@/types/global'
import { useUserInfoStore } from '@/stores'
/**
 * 获取有效客户|供应商列表
 * @param clientType 客户类型
 */
export const getValidSupplierClientAPI = (clientType: ClientTypeEnum) => {
  return http<GetValidSupplierClientDto[]>({
    method: 'GET',
    url: `/api/mes-manager/SupplierClient/get-open-client`,
    data: {
      clientType,
    },
  })
}

/**
 * 查询供应商|客户列表
 * @param input query
 */
export const queryPageSupplierClientApi = (input: any) => {
  return http<PageResult<any>>({
    method: 'GET',
    url: `/api/mes-manager/SupplierClient/query`,
    data: input,
  })
}

/**
 * 创建供应商|客户
 * @param input query
 */
export const createAndUpdateSupplierClientApi = (input: any) => {
  const useUserInfo = useUserInfoStore()
  return http<any>({
    method: 'POST',
    url: `/api/mes-manager/SupplierClient/create-and-update`,
    data: {
      ...input,
      clientStatus: PublicStatusEnum.正常,
      companyId: useUserInfo.userInfo?.companyId,
    },
  })
}

/**
 * 禁用
 * @param id key
 */
export const banSupplierClientApi = (id: string) => {
  return http<any>({
    method: 'DELETE',
    url: `/api/mes-manager/SupplierClient/ban/${id}`,
  })
}

/**
 * 启用
 * @param id key
 */
export const openSupplierClientApi = (id: string) => {
  return http<any>({
    method: 'POST',
    url: `/api/mes-manager/SupplierClient/open/${id}`,
  })
}
